src/components/CreateMatchExperienceModal/index.tsx

```diff
@@ -0,0 +1,252 @@
+        mutationFn: async (file: File) => {
+            const formData = new FormData();
+            formData.append('file', file);
+            const { data } = await FileService.handleUpload(formData);
+            return data;
+        },
+        onSuccess: (data) => {
+            if (data.url) {
+                message.success('Image uploaded successfully');
+                const filePath = data.url.split('/public/')[1];
+                setImageUrl(filePath);
+                setValue('picture', filePath);
+            } else {
+                message.error('Upload failed');
+            }
+        },
+        onError: () => {
+            message.error('Upload error');
+        },
+    });
+
+    const { data: countries = [] } = fetchData('countries', isOpen, SoccerService.getCountries);
+    const { data: leagues = [] } = fetchData('leagues', selectedCountry, SoccerService.getLeagues);
+    const { data: stadiums = [] } = fetchData('stadiums', selectedCountry, SoccerService.getVenues);
+    const { data: teams = [] } = fetchData('teams', selectedLeague, SoccerService.getTeams);
+
+    const {
+        control,
+        handleSubmit,
+        setValue,
+        reset,
+        formState: { errors },
+    } = useForm<CreateMatchExperienceModalValues>({
+        resolver: yupResolver(MatchExperienceSchema),
+    });
+
+    const onSubmit = async (values: CreateMatchExperienceModalValues) => {
+        // const { title, description, country, date, stadium, league, homeTeam, awayTeam } = values;
+        try {
+            await MatchExperienceService.createMatchExperience({
+                ...values,
+                picture: imageUrl,
+                createdBy: '123412341234123412341234',
+            });
+
+            message.success('MatchExperience created successfully');
+
+            reset();
+            setImageUrl('');
+            setSelectedCountry('');
+            setSelectedLeague(undefined);
+            onClose();
+        } catch (error) {
+            message.error('An error occurred while submitting the matchExperience.');
+        }
+    };
+
+    return (
+        <Modal
+            title="Create MatchExperience"
+            open={isOpen}
+            onCancel={() => {
+                reset();
+                setImageUrl('');
+                setSelectedCountry('');
+                setSelectedLeague(undefined);
+                onClose();
+            }}
+            footer={null}
+            className={styles.modal}
+        >
+            <Form layout="vertical" className={styles.form} onFinish={handleSubmit(onSubmit)}>
+                <Form.Item
+                    label="Country"
+                    validateStatus={errors.country ? 'error' : ''}
+                    help={errors.country?.message}
+                >
+                    <AutoComplete
+                        options={countries.map((option) => ({ value: option.name }))}
+                        onSelect={(value) => {
```

Comment:
would make a `formStateCleanup` function in the component that does the same thing and just use it here instead of inline:

```suggestion
onSelect={formStateCleanup}
```